LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -924,6 +1022,62 @@ extension DoseStore {
             completion(.success(reconciledDoses))
         }
     }
+    
+    /// *This method should only be called from within a managed object context block.*
+    ///
+    /// - Parameter startDate: The earliest outside dose event date to include
+    /// - Returns: An array of logged dose managed objects, in reverse-chronological order
+    /// - Throws: An error describing the failure to fetch objects
+    public func getLoggedDoses(since startDate: Date, completion: @escaping (_ result: DoseStoreResult<[PersistedOutsideDose]>) -> Void) {
+        persistenceController.managedObjectContext.perform {
+            do {
+                let events = try self.getLoggedDoses(
+                    matching: NSPredicate(format: "startDate >= %@", startDate as NSDate),
+                    chronological: false
+                )
+
+                completion(.success(events))
+            } catch let error as DoseStoreError {
+                completion(.failure(error))
+            } catch {
+                assertionFailure()
+            }
+        }
+
+
+    }
+
+    /// *This method should only be called from within a managed object context block.*
+    ///
+    /// Objects are ordered by date using the DoseType sort ordering as a tiebreaker for stability
+    ///
+    /// - Parameters:
+    ///   - predicate: The predicate to apply to the objects
+    ///   - chronological: Whether to return the objects in chronological or reverse-chronological order
+    /// - Returns: An array of pump events in the specified order by date
+    /// - Throws: An error describing the failure to fetch objects
+    private func getLoggedDoses(matching predicate: NSPredicate, chronological: Bool, limit: Int? = nil) throws -> [PersistedOutsideDose] {
```

Comment:
[nit] see above: consider `getStoredDoses`
[nit 2] `chronological` seems a poor term for this Bool argument...consider removing if not needed, or else use an enum

Reply:
This matches existing terminology and usage of another long-standing method in DoseStore.